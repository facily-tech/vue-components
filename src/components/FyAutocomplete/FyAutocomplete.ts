/// <reference path="../../../node_modules/vuetify/src/globals.d.ts" />
import Vue from 'vue';

import { VAutocomplete } from 'vuetify/lib';

const base = Vue.extend({ mixins: [VAutocomplete] });

type options = InstanceType<typeof base>;

export default base.extend<options>().extend({
  name: 'fy-autocomplete',
  computed: {
    class(): Record<string, unknown> {
      return {
        ...VAutocomplete.options.computed.classes.call(this),
        'fy-autocomplete': true,
      };
    },
  },
});
