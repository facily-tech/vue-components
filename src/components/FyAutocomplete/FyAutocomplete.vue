<template>
  <v-autocomplete dense outlined v-bind="$attrs" v-on="$listeners">
    <slot v-for="(_, name) in $slots" :name="name" :slot="name" />
  </v-autocomplete>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
  name: 'FyAutocomplete',
});
</script>
