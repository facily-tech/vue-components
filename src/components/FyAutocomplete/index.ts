import FyAutocomplete from './FyAutocomplete.vue';

export { FyAutocomplete };
export default FyAutocomplete;
