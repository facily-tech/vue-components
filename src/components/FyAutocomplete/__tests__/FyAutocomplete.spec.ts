import Vue from 'vue';
import Vuetify from 'vuetify';
import FyAutocomplete from '../FyAutocomplete.vue';
import { mount, Wrapper } from '@vue/test-utils';

Vue.use(Vuetify);

describe('FyAutocomplete', () => {
  type Instance = InstanceType<typeof FyAutocomplete>;
  let mountFunction: (options?: Record<string, unknown>) => Wrapper<Instance>;

  beforeEach(() => {
    document.body.setAttribute('data-app', 'true');

    mountFunction = (options = {}) => {
      return mount(FyAutocomplete, {
        ...options,
        vuetify: new Vuetify(),
      });
    };
  });

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });
});
