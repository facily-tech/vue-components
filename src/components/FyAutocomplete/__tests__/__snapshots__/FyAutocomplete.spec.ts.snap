// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FyAutocomplete should render component and match snapshot 1`] = `
<div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete">
  <div class="v-input__control">
    <div role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-2" class="v-input__slot">
      <fieldset aria-hidden="true">
        <legend style="width: 0px;"><span class="notranslate">â€‹</span></legend>
      </fieldset>
      <div class="v-select__slot"><input id="input-2" type="text" autocomplete="off">
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
        </div><input type="hidden">
      </div>
      <div class="v-menu">
        <!---->
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
